test/utils/test_ast.jl

```diff
@@ -309,6 +309,177 @@ end
     end
 end
 
+@testset "token_at_offset / token_before_offset" begin
+    # Test token_at_offset with simple identifiers
+    let code = "alpha beta gamma"
+        ps = parsedstream(code)
+        @test String(ps.textbuf[JETLS.byte_range(@something(JETLS.token_at_offset(ps, 1)))]) == "alpha"
+        @test String(ps.textbuf[JETLS.byte_range(@something(JETLS.token_at_offset(ps, 3)))]) == "alpha"
+        @test String(ps.textbuf[JETLS.byte_range(@something(JETLS.token_at_offset(ps, 5)))]) == "alpha"
+        @test String(ps.textbuf[JETLS.byte_range(@something(JETLS.token_at_offset(ps, 6)))]) == " "
+        @test String(ps.textbuf[JETLS.byte_range(@something(JETLS.token_at_offset(ps, 8)))]) == "beta"
+        @test isnothing(JETLS.token_at_offset(ps, 100))
+    end
+    let code = "foo(bar)"
+        ps = parsedstream(code)
+        @test String(ps.textbuf[JETLS.byte_range(@something(JETLS.token_at_offset(ps, 2)))]) == "foo"
+        @test String(ps.textbuf[JETLS.byte_range(@something(JETLS.token_at_offset(ps, 4)))]) == "("
+        @test String(ps.textbuf[JETLS.byte_range(@something(JETLS.token_at_offset(ps, 6)))]) == "bar"
+    end
+
+    # Test token_before_offset
+    let code = "a + b"
+        ps = parsedstream(code)
+        @test String(ps.textbuf[JETLS.byte_range(@something(JETLS.token_before_offset(ps, 2)))]) == "a"
+        @test String(ps.textbuf[JETLS.byte_range(@something(JETLS.token_before_offset(ps, 3)))]) == " "
+        @test String(ps.textbuf[JETLS.byte_range(@something(JETLS.token_before_offset(ps, 4)))]) == "+"
+        @test JETLS.token_before_offset(ps, 1) !== nothing  # COMBAK should probably return nothing instead
```

Comment:
I think we should remove `TOMBSTONE` from the token list that `TokenCursor` returns.
In this case, returning `nothing` seems more natural, and we should also be able to remove the special case for `prev_nontrivia`.